adapters/acuityads/acuityads.go

```diff
@@ -0,0 +1,193 @@
+}
+
+func GetHeaders(request *openrtb.BidRequest) *http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("X-Openrtb-Version", "2.5")
+
+	if request.Device != nil {
+		if len(request.Device.UA) > 0 {
+			headers.Add("User-Agent", request.Device.UA)
+		}
+
+		if len(request.Device.IPv6) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IPv6)
+		}
+
+		if len(request.Device.IP) > 0 {
+			headers.Add("X-Forwarded-For", request.Device.IP)
+		}
+	}
+
+	return &headers
+}
+
+func (a *AcuityAdsAdapter) MakeRequests(
+	openRTBRequest *openrtb.BidRequest,
+	reqInfo *adapters.ExtraRequestInfo,
+) (
+	requestsToBidder []*adapters.RequestData,
+	errs []error,
+) {
+
+	var errors []error
+	var acuityAdsExt *openrtb_ext.ExtAcuityAds
+	var err error
+
+	for i, imp := range openRTBRequest.Imp {
+		acuityAdsExt, err = a.getImpressionExt(&imp)
+		if err != nil {
+			errors = append(errors, err)
+			break
+		}
+		openRTBRequest.Imp[i].Ext = nil
+	}
+
+	if len(errors) > 0 {
+		return nil, errors
+	}
+
+	url, err := a.buildEndpointURL(acuityAdsExt)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	reqJSON, err := json.Marshal(openRTBRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return []*adapters.RequestData{{
+		Method:  http.MethodPost,
+		Body:    reqJSON,
+		Uri:     url,
+		Headers: *GetHeaders(openRTBRequest),
+	}}, nil
+}
+
+func (a *AcuityAdsAdapter) getImpressionExt(imp *openrtb.Imp) (*openrtb_ext.ExtAcuityAds, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}
+	}
+	var acuityAdsExt openrtb_ext.ExtAcuityAds
+	if err := json.Unmarshal(bidderExt.Bidder, &acuityAdsExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}
+	}
+	return &acuityAdsExt, nil
+}
+
+func (a *AcuityAdsAdapter) buildEndpointURL(params *openrtb_ext.ExtAcuityAds) (string, error) {
+	endpointParams := macros.EndpointTemplateParams{Host: params.Host, AccountID: params.AccountID}
+	return macros.ResolveMacros(a.endpoint, endpointParams)
+}
+
+func (a *AcuityAdsAdapter) CheckResponseStatusCodes(response *adapters.ResponseData) error {
+	if response.StatusCode == http.StatusNoContent {
+		return &errortypes.BadInput{Message: "No bid response"}
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: [ %d ]", response.StatusCode),
+		}
+	}
+
+	if response.StatusCode == http.StatusServiceUnavailable {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("Something went wrong, please contact your Account Manager. Status Code: [ %d ] ", response.StatusCode),
+		}
+	}
+
+	if response.StatusCode < http.StatusOK || response.StatusCode >= http.StatusMultipleChoices {
```

Comment:
Just double checking: function `CheckResponseStatusCodes` doesn't seem to consider status codes other than: 
 - `StatusNoContent            = 204 // RFC 7231, 6.3.5`
 - `StatusBadRequest           = 400 // RFC 7231, 6.5.1`
 - `StatusServiceUnavailable   = 503 // RFC 7231, 6.6.4`
 - `StatusCreated              = 201 // RFC 7231, 6.3.2`
 - `StatusAccepted             = 202 // RFC 7231, 6.3.3`
 - `StatusNonAuthoritativeInfo = 203 // RFC 7231, 6.3.4`
 - `StatusResetContent         = 205 // RFC 7231, 6.3.6`
 - `StatusPartialContent       = 206 // RFC 7233, 4.1`
 - `StatusMultiStatus          = 207 // RFC 4918, 11.1`
 - `StatusAlreadyReported      = 208 // RFC 5842, 7.1`
 - `StatusIMUsed               = 226 // RFC 3229, 10.4.1`
 - `StatusMultipleChoices      = 300 // RFC 7231, 6.4.1`

Other adapters do:
```
	if response.StatusCode != http.StatusOK {
		return nil, []error{&errortypes.BadServerResponse{
			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
		}}
	}
```
To cover everything else. Would that be useful/desired in our case?